File: backend/app/schemas/transfers.py
from typing import Optional
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel
from app.models.enums import TransferStatus

class TransferRequestBase(BaseModel):
    reason: str
    target_bed_id: Optional[UUID] = None

class TransferRequestCreate(TransferRequestBase):
    contract_id: UUID

class TransferRequestResponse(TransferRequestBase):
    id: UUID
    student_id: UUID
    contract_id: UUID
    status: TransferStatus
    admin_response: Optional[str] = None
    created_at: datetime
    student_name: Optional[str] = None
    student_code: Optional[str] = None
    current_room_name: Optional[str] = None
    current_bed_label: Optional[str] = None
    target_room_name: Optional[str] = None
    target_bed_label: Optional[str] = None
    
    class Config:
        from_attributes = True

class TransferRequestUpdate(BaseModel):
    status: TransferStatus
    admin_response: Optional[str] = None
    assigned_bed_id: Optional[UUID] = None

File: frontend/src/types/transfers.ts
import { UUID } from "./common";

export enum TransferStatus {
    PENDING = "CHO_DUYET",
    APPROVED = "DA_DUYET",
    REJECTED = "TU_CHOI",
    COMPLETED = "HOAN_TAT", // Frontend only? Or maybe backend maps it? Keeping for safety but vietnamized.
    CANCELLED = "DA_HUY"
}

export interface TransferRequest {
    id: UUID;
    student_id: UUID;
    contract_id: UUID;
    target_bed_id?: UUID;
    reason: string;
    status: TransferStatus;
    admin_response?: string;
    created_at: string;

    // Enriched UI fields
    student_name?: string;
    student_code?: string;
    current_room_name?: string;
    current_bed_label?: string;
    target_room_name?: string;
    target_bed_label?: string;
}

export interface TransferRequestCreate {
    contract_id: UUID;
    target_bed_id?: UUID;
    reason: string;
}

export interface TransferRequestUpdate {
    status: TransferStatus;
    admin_response?: string;
    assigned_bed_id?: string;
}
